// realtime-data/index.js
const Waterline     = require('waterline');
const mysqlAdapter  = require('sails-mysql');
const userCategory  = require('./models/user-category.js');
const user          = require('./models/user');
const space         = require('./models/space');
const spaceRoom     = require('./models/space-room');
const roomMessage   = require('./models/room-message');
const roomUser      = require('./models/room-user');

class Data {

  // mysql
  // db
  // config
  // collections
  // connections

  constructor (settings) {

    this.mysql = settings.mysql;

    this.db = new Waterline();

    this.db.loadCollection(user);
    this.db.loadCollection(userCategory);
    this.db.loadCollection(space);
    this.db.loadCollection(spaceRoom);
    this.db.loadCollection(roomMessage);
    this.db.loadCollection(roomUser);
  }

  init (callback) {

    const m = this.mysql;
    this.config = {
      adapters: {
        mysql: mysqlAdapter,
      },
      connections: {
        diamante: {
          adapter: 'mysql',
          url: `mysql2://${m.user}:${m.pass}@${m.host}:${m.port}/${m.db}`,
        },
      },
    };

    return this.db.initialize(this.config, (err, models) => {
      if (err) {
        if (callback) {
          callback(err, models);
        }
        else {
          throw err;
        }
      }
      else {
        this.models = models.collections;
        this.connections = models.connections;
        callback(err, models);
      }
    });
  }
}

module.exports = Data;
